### PR TITLE
[*] Revamped `MySQLDatabaseGateway` and other major modifications to use cases

### PR DESCRIPTION
**Pull request by Lei-Tin**
Refer to this [issue](https://github.com/CSC207-2022F-UofT/course-project-bill-gates/issues/75) for the reason of this PR. 

# Database class modifications
- Completely removed `QueryEntryData` and all the related classes with that, also removed `QueryBillData` with it
- Modified `EntryBuilder` so that the date is truncated to seconds
- Moved `QueryUserData` into entities
- Modified `DatabaseGateway` so that it aligns with the new descriptions
- Modified tests so that it works with the new modification

# Use Case modifications
Fixed all use case issues. 

Now that all use cases work with the new `MySQLDatabaseGateway` modifications

### CODE REVIEW

File: src/main/java/billgates/entities/EntryBuilder.java [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
 import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
 
 /**
  * Clean Architecture Layer: Enterprise Business Rules
  * This is a builder class that help to build an Entry or Splitter Entry.
- * This class implements a builder design pattern.
  * <p>
+ * Design Pattern: Builder
```
Comment (ScottCTD) [inline src/main/java/billgates/entities/EntryBuilder.java:10]: You could move this line to the following line of `Clean Architecture....`

File: src/main/java/billgates/entities/User.java [hunk 2]
```diff
@@ -6,7 +6,8 @@
  * Clean Architecture Layer: Enterprise Business Rules
  * This class represents a user entity in our program.
  * Only one user is allowed in the running time of our program.
- * We implement the User using singleton design pattern.
+ * <p>
+ * Design Pattern: Singleton
```
Comment (ScottCTD) [inline src/main/java/billgates/entities/User.java:10]: Also here. You MAY do that. It's okay.

**Review by ScottCTD — APPROVED** [review]
Great refactors!

**Review by xgao28 — APPROVED** [review]
Nice

**Review by ScottCTD — APPROVED** [review]
There are still tons of conflicts with newer PRs to be fixed. 

However, let's merge this pr first.

**Pull request merged**